### ISSUE TITLE
Problems with apprise

### ISSUE THREAD

**Issue by vince-p** [auanasgheps/snapraid-aio-script#137]
**Describe the bug**
I cannot get the script to access apprise correctly.
There are two problems:

Problem 1:
The script always says "apprise has not been found and will be installed... apprise installed successfully."
This happens even if the script has run previously and also this still happens after I manually install apprise.

Problem 2:
The value for APPRISE_BIN set in the config file does not get passed on to the main script.
This means that none of the APPRISE lines work.

**To Reproduce**
As a test I add the line "echo $APPRISE_BIN" near the top of the main script. The echo command always returns blank.
when I run ' command -v apprise' from the command line, the output is "/home/media/.local/bin/apprise"
So apprise is installed but the APPRISE_BIN variable is not passing through to the main script.

If change the value of APPRISE_BIN to "/home/media/.local/bin/apprise"
Then I get the following error when the script runs:

```
Sending notification using Apprise service(s).
Traceback (most recent call last):
  File "/home/media/.local/bin/apprise", line 5, in <module>
    from apprise.cli import main
ModuleNotFoundError: No module named 'apprise'
```

I know that apprise is working outside of the script. Eg when I run the following line from the command line it works fine:
apprise -vv -t 'TEST TITLE' -b 'BODY' APPRISE_URL

**Please complete the following information:**
 - Distro: Ubuntu 22.04
 - AIO Script Version 3.4 Beta 2

**Comment by auanasgheps** [comment]
Hi, thanks for reporting an issue on this new feature, I was looking forward to see something like this.

Issue 1.
The script checks if apprise is installed by executing `dpkg-query -W -f='${Status}' apprise`
What happens on your server if you run this manually?

Also, the scripts attemps to use the apprise binary by executing `command -v apprise`
What happens on your server if you run this manually?

Also, what version of Apprise are you using? For example I've noticed that Debian has very, very old Apprise version in their repo. I might turn off automatic Apprise installation if I can't find an easy way to install the latest.

**Comment by auanasgheps** [comment]
I've found the solution to the version issue. I'll be using pip to do this, which will be installed to install apprise. I know it's not super efficient but it's the only way to ensure a recent version of apprise is installed. When I started testing it, I started with the one in the Debian repo and it was super old.

**Comment by vince-p** [comment]
Glad you have the install issue figured out!
When i enter the "command -v apprise" in the terminal, it returns the folder where apprise is installed on my system ("/home/media/.local/bin/apprise"). Nothing else happens.
I'm using apprise version 1.9.1

**Comment by auanasgheps** [comment]
Have you installed apprise using pipx?
Execute `pipx ensurepath` and try again.

**Comment by vince-p** [comment]
Hi, sorry for the slow reply. I didn't install apprise with pipx so that command doesnt work for me.

**Comment by auanasgheps** [comment]
Ok, can you try this approach?
Remove your apprise installation.

Install `python3-pipx` using your package manager

Run `pipx ensurepath`
Finally install apprise via pipx
`pipx install apprise`

**Comment by auanasgheps** [comment]
Hi, any news?

**Comment by vince-p** [comment]
ok sorry for the very long delay.
I removed apprise and reinstalled using pipx as you suggested.

When the script is run, it now includes the following near the top of the output:
`Apprise service notification is enabled.
apprise has not been found and will be installed...
apprise installed successfully.
`

This message is odd as apprise WAS already installed using pipx before running the script.
In any case, the script does now seem to work and I get apprise notifications as expected.

**Comment by auanasgheps** [comment]
Thanks for the feedback! Does the message appear at every run?

**Comment by vince-p** [comment]
Yes I get the same message every time. It says it is not installed (even though I have installed using pipx).
But the notifications do work.

**Comment by auanasgheps** [comment]
what does show the command `pipx list` on your server?

**Comment by auanasgheps** [comment]
Ok, I think I found the issue. It's because I made an enhancement AFTER the beta 2. Can you please try the script in the `dev `branch, it should detect it properly!

**Comment by vince-p** [comment]
I tried with the script version in the dev branch but had some errors.
Here's the output:

`SnapRAID Script Job started [Mon 21 Apr 17:25:30 AEST 2025]
Running SnapRAID version 12.2
SnapRAID AIO Script version 3.4
Using configuration file: ./script-config.conf
----------------------------------------
## Preprocessing
Healthchecks.io notification is enabled. Notifications sent to https://hc-ping.com/.
Apprise service notification is enabled.
pipx is not installed. Installing pipx...
Hit:1 http://ubuntu.mirror.serversaustralia.com.au/ubuntu jammy InRelease
Hit:2 http://ubuntu.mirror.serversaustralia.com.au/ubuntu jammy-updates InRelease
Hit:3 http://ubuntu.mirror.serversaustralia.com.au/ubuntu jammy-backports InRelease
Hit:4 http://ubuntu.mirror.serversaustralia.com.au/ubuntu jammy-security InRelease
Hit:5 https://download.docker.com/linux/ubuntu jammy InRelease
Get:6 https://pkgs.tailscale.com/stable/ubuntu jammy InRelease
Ign:7 https://download.webmin.com/download/newkey/repository stable InRelease
Hit:8 https://download.webmin.com/download/newkey/repository stable Release
Hit:9 https://ppa.launchpadcontent.net/awesometic/ppa/ubuntu jammy InRelease
Fetched 6,578 B in 2s (3,829 B/s)
Reading package lists...
./snapraid-aio-script.sh: line 1170: pipx: command not found
./snapraid-aio-script.sh: line 1174: pipx: command not found
apprise is not installed. Installing apprise using pipx...
./snapraid-aio-script.sh: line 1176: pipx: command not found
Error: Failed to install apprise.
SnapRAID is not running, proceeding.
SnapRAID output retention is enabled. Detailed logs will be kept in /home/media/snapraid/logs for 1 days.
Checking if all parity and content files are present.
All parity files found.
All content files found.
Stopping the script because the previous SnapRAID sync did not complete correctly.
Sending notification using Apprise service(s).
./snapraid-aio-script.sh: line 1036: [: : integer expression expected
Email address is set. Sending email report to **[REDACTED]** [Mon 21 Apr 17:25:53 AEST 2025]
./snapraid-aio-script.sh: line 952: [: : integer expression expected
`

**Comment by auanasgheps** [comment]
There are two issues here.

You see some "integer expression expected" because you need to update the config file. I added some new options.

The real issue however is that the script is unable to install pipx on your system. I believe I need to adjust the install command.

**Comment by auanasgheps** [comment]
Currently the command is `sudo apt-get install -y python3-pipx`
Try to execute it manually, I believe will throw an error.

In that case, try with `sudo apt-get install -y pipx`. If this works, execute `pipx ensurepath`, then you can run the script again.

I'll update the script after your feedback.

**Comment by vince-p** [comment]
Sorry for the very slow response.
You are correct - " sudo apt-get install -y python3-pipx" does lead to the
following error:

Reading package lists... Done
Building dependency tree... Done
Reading state information... Done
E: Unable to locate package python3-pipx

**Comment by vince-p** [comment]
Sorry you can ignore my last message entirely - the error was nothing to do with your script, I was just running out of space on the array which had prevented the previous sync process was completing. I've since freed up some space and re-synced, and now your script runs perfectly.
